### PR TITLE
Remove unused runtime library path on non-Darwin platforms

### PR DESCRIPTION
**Pull request by finagolfin**
Fixes apple/swift#63416

### CODE REVIEW

**Comment by finagolfin** [comment]
@stevapple and @kateinoigakukun, please let me know if this is okay on Windows and Wasm too.

File: Sources/SwiftDriver/Jobs/Toolchain+LinkerSupport.swift [hunk 1]
```diff
@@ -46,7 +46,8 @@ extension Toolchain {
       result.append(path)
     }
 
-    if let sdkPath = sdkPath.map(VirtualPath.lookup) {
+    // Only Darwin places libraries directly in /sdk/usr/lib/swift/.
+    if triple.isDarwin, let sdkPath = sdkPath.map(VirtualPath.lookup) {
```
[the remark this answers is not in the document]
Comment (finagolfin) [reply Sources/SwiftDriver/Jobs/Toolchain+LinkerSupport.swift:50]: As in, the Windows toolchain places runtime dlls directly in the resource directory? I thought you place them in arch-specific directories like `/sdk/usr/lib/swift/windows/x86_64/`?

[the remark this answers is not in the document]
Comment (stevapple) [reply Sources/SwiftDriver/Jobs/Toolchain+LinkerSupport.swift:50]: Is this for `.lib`s only?🤔️ I thought I’ve misunderstood then. Windows SDK puts `.swiftmodule` bundles here, and all libraries are in arch-specific directories.

Runtime DLLs are kept separately. The compiler don’t need them to perform dynamic linking.

[the remark this answers is not in the document]
Comment (finagolfin) [reply Sources/SwiftDriver/Jobs/Toolchain+LinkerSupport.swift:50]: Yes, all this does is add `-L /sdk/usr/lib/swift/` on non-Windows, so if there is nothing to link against there, we can remove that flag.

I just checked on Windows, and this method is unused by that toolchain, so this pull makes no difference on Windows. However, [the C++ Driver equivalent _does_ use that method and looks in `/sdk/usr/lib/swift/x86_64/`](https://github.com/apple/swift/blob/48de17d1408323d38b8b4ee981ffe22254e95dad/lib/Driver/WindowsToolChains.cpp#L124), which probably doesn't even exist, so let me know if that other pull is fine.

[the remark this answers is not in the document]
Comment (stevapple) [reply Sources/SwiftDriver/Jobs/Toolchain+LinkerSupport.swift:50]: I’d suggest adding both `-L /sdk/usr/lib/swift/<os>/<arch>/` and `-L /sdk/usr/lib/swift/<os>/`, and always looking in the former first. I remembered that there should be nothing different now regarding Windows and Linux library searching🤔️ Darwin is the only exception.

**Comment by finagolfin** [comment]
@artemcm, this is the equivalent of my C++ Driver pull you just approved, would you test and merge both?

**Comment by artemcm** [comment]
@swift-ci test

**Comment by finagolfin** [comment]
@artemcm, this one and the legacy driver backport linked above have passed all tests, just need you to approve and merge both.

**Comment by finagolfin** [comment]
@artemcm, the upstream version of this pull that you approved has been merged, just need you to approve and merge this one too.

**Comment by artemcm** [comment]
@swift-ci test Windows platform

**Pull request merged**